server/pbench/bin/pbench-dispatch

```diff
@@ -0,0 +1,163 @@
+#! /bin/bash
+# -*- mode: shell-script -*-
+
+# This script is the first part of the pipeline that processes pbench
+# results tarballs.
+
+# First stage:  pbench-dispatch looks in all the TODO
+#               directories, checks MD5 sums and
+#               creates symlinks in all the state directories for
+#               the particular environment where this server runs
+#               (e.g. for the production environment, links are created
+#               in the TO-UNPACK, TO-INDEX, TO-COPY-SOS and TO-BACKUP
+#               directories; for a satellite, it will only create links
+#               in the TO-UNPACK and TO-SYNC directories). Then the
+#               symlink in TODO is deleted: we don't want to deal with
+#               this tarball again; but if there are errors, we may
+#               keep it in TODO and try again, if the error is recoverable.
+#               Any errors are reported for possible action by an admin.
+#               
+
+# assumptions:
+# - this script runs as a cron job
+# - tarballs and md5 sums are uploaded by pbench-move/copy-results to
+#   $ARCHIVE/$(hostname -s) area.
+# - pbench-move/copy-results also makes a symlink to each tarball it uploads
+#   in $ARCHIVE/TODO, thereby telling this script to process the tarball.
+
+# This script loops over the contents of $archive/<hostname>/TODO, verifies the md5
+# sum of each tarball, and if correct, it dispatches the tarball for further
+# processing.
+
+
+###########################################################################
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)
+        PROG=$(basename $0)
+        ;;
+esac
+
+. $dir/pbench-base.sh
+###########################################################################
+
+log_init $(basename $0)
+
+# check that all the directories exist
+test -d $ARCHIVE || doexit "Bad ARCHIVE=$ARCHIVE"
+test -d $UNPACK_DIR || doexit "Bad UNPACK=$UNPACK_DIR"
+
+# make sure only one copy is running.
+# Use 'flock -n $LOCKFILE /path/to/pbench-dispatch' in the
+# crontab to ensure that only one copy is running. The script itself
+# does not use any locking.
+
+# the link source and destination for this script
+linksrc=TODO
+linkdestlist=$(getconf.py -l dispatch-states pbench-server)
```

Comment:
If linkdestlist is empty, the script should bail with an error. This is most likely to be caused by a config file error (no dispach-states defined - or a typo), but the effect is to delete the TODO links without doing anything else.